include/libtorrent/announce_entry.hpp

```diff
@@ -62,13 +91,7 @@ namespace libtorrent
 		// trackerid is sent).
 		std::string trackerid;
 
-		// if this tracker has returned an error or warning message
-		// that message is stored here
-		std::string message;
-
-		// if this tracker failed the last time it was contacted
-		// this error code specifies what error occurred
-		error_code last_error;
+		std::vector<announce_endpoint> endpoints;
```

Comment:
I think I've found a reasonably clean solution for this. Conceptually each (torrent, listen socket) pair is treated like a separate torrent which does it's own traversal of the tracker list. For efficiency reasons the code traverses all listen sockets for a single torrent at once while keeping a vector of states for each socket.